pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -180,9 +180,8 @@ def _run_optimization(
         result = AttributeDict()
         closure = self._make_closure(split_batch, batch_idx, opt_idx, optimizer, self._hiddens, result)
 
-        if self.trainer.fit_loop.should_accumulate():
-            # For gradient accumulation
-
+        # when the training type plugin handles accumulation, we want to always call the optimizer step
+        if not self.trainer.accelerator.handles_accumulate_grad_batches and self.trainer.fit_loop.should_accumulate():
```

Comment:
Can we move this logic under `should_accumulate()`. For loop customization, users should not have to know training type plugin handles accumulation differently.

Reply:
We can't because we need to call the `optimizer.step` if the TTP handles it as they also track whether accumulation should be applied.